### PR TITLE
Change spawning rules for airborne players and rebel UAVs

### PR DESCRIPTION
**Pull request by jaj22**
## What type of PR is this.
1. [ ] Bug
2. [X] Change
3. [ ] Enhancement

### What have you changed and why?
1. Excluded additional units in the same vehicle from spawning. Perf difference isn't large in most cases, but it doesn't hurt either.

2. Excluded rebels in fast-moving fixed wing aircraft from spawning. This is to prevent CAS/CAP rebels from repeatedly spawning and despawning locations that they overfly. If they're playing solo and need a location spawned then they can simply move an HC squad within range.

3. Included rebel UAVs in spawn list. This prevents an exploit where you could park UAVs with attached explosives near a flag and then spawn it in.

4. Excluded airborne+moving rebels from spawning civilian vehicles in towns and rebel garrisons. You don't care about those when airborne so they're just a performance cost.

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.
1. [X] Have you loaded the mission in LAN host?
4. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [X] No
3. [ ] Yes (Please provide further detail below.)

### CODE REVIEW

File: A3A/addons/core/functions/Base/fn_distance.sqf [hunk 1]
```diff
@@ -409,25 +409,25 @@ do
...
-        _players = _players select { _x getVariable ["owner",objNull] == _x };
+        // only count one spawner per vehicle
+        _occupants = units Occupants select { _x getVariable ["spawner", false] and _x == effectiveCommander vehicle _x };
+        _invaders = units Invaders select { _x getVariable ["spawner", false] and _x == effectiveCommander vehicle _x };
+
+        // Exclude players in fast-moving fixed-wing aircraft
+        _teamplayer = units teamPlayer select {
+            private _veh = vehicle _x;
+            _x getVariable ["spawner", false] and _x == effectiveCommander _veh
+            and (_veh == _x or {!(_veh isKindOf "Plane" and speed _veh > 250)})
```
Comment (killerswin2) [inline A3A/addons/core/functions/Base/fn_distance.sqf:420]: 250 seems a little dicey. Do we have a minimum speed set for spawns?

Comment (jaj22) [reply A3A/addons/core/functions/Base/fn_distance.sqf:420]: Not sure what you mean.

Comment (killerswin2) [reply A3A/addons/core/functions/Base/fn_distance.sqf:420]: Uavs/ppanes can be slow, specially on cup. The speed check concerns me that some uavs/planes won't receive these performance gains

Comment (jaj22) [reply A3A/addons/core/functions/Base/fn_distance.sqf:420]: Rebel UAVs spawn regardless. Fast UAVs would be more of a problem, although I don't think there are any likely cases. They're difficult to steal at least, and not very useful.

250km/h is 134 knots, which is below stalling speed for most fixed-wing jets. At worst I think you'd need to be flying intentionally slowly, in which case spawning is fine. Some non-jets can reasonably fly slower than that, but in that case it's also enough time for a garrison to spawn in and shoot them down with 50cal.

Comment (killerswin2) [reply A3A/addons/core/functions/Base/fn_distance.sqf:420]: Okay, that works for me.

**Review by killerswin2 — APPROVED** [review]
Looks good to me

**Review by Bob-Murphy — APPROVED** [review]
lgtm

**Pull request merged**